File: apps/aiii-desktop/src-tauri/src/commands/mcp.rs
//! MCP server Tauri commands
//!
//! Provides commands for controlling the MCP server from the UI.

use tauri::State;

use crate::mcp::state::McpStatus;
use crate::mcp::McpServerState;

/// Start the MCP server
///
/// Starts the MCP server using stdio transport, making it available
/// to AI applications like Claude Desktop.
///
/// # Returns
/// Ok(()) on success, Err with message on failure
#[tauri::command]
pub async fn mcp_start(mcp_state: State<'_, McpServerState>) -> Result<(), String> {
    mcp_state.start().await
}

/// Stop the MCP server
///
/// Gracefully stops the running MCP server.
///
/// # Returns
/// Ok(()) on success, Err with message on failure
#[tauri::command]
pub async fn mcp_stop(mcp_state: State<'_, McpServerState>) -> Result<(), String> {
    mcp_state.stop().await
}

/// Get the MCP server status
///
/// Returns the current status of the MCP server including whether it's
/// running and what transport it's using.
///
/// # Returns
/// McpStatus containing server state information
#[tauri::command]
pub async fn mcp_status(mcp_state: State<'_, McpServerState>) -> Result<McpStatus, String> {
    Ok(mcp_state.status().await)
}

File: apps/desktop/src/lib/api/mcp.ts
/**
 * MCP API - Tauri IPC wrappers for MCP server operations
 */

import { invoke } from "@tauri-apps/api/core"

/**
 * MCP server status
 */
export interface McpStatus {
  /** Whether the server is currently running */
  running: boolean
  /** Port number if using HTTP transport (None for stdio) */
  port: number | null
  /** Transport type being used */
  transport: string
}

/**
 * Start the MCP server
 *
 * Starts the MCP server using stdio transport, making it available
 * to AI applications like Claude Desktop.
 *
 * @throws Error if server is already running or vault is locked
 */
export const mcpStart = (): Promise<void> => invoke("mcp_start")

/**
 * Stop the MCP server
 *
 * Gracefully stops the running MCP server.
 *
 * @throws Error if server is not running
 */
export const mcpStop = (): Promise<void> => invoke("mcp_stop")

/**
 * Get the MCP server status
 *
 * @returns Current server status including running state and transport
 */
export const mcpStatus = (): Promise<McpStatus> => invoke("mcp_status")
